pkgs/applications/version-management/git-and-tools/git/default.nix

```diff
@@ -78,7 +78,7 @@ stdenv.mkDerivation {
     ++ lib.optionals withLibsecret [ pkg-config glib libsecret ];
 
   # required to support pthread_cancel()
-  NIX_LDFLAGS = lib.optionalString (!stdenv.cc.isClang) "-lgcc_s"
+  NIX_LDFLAGS = lib.optionalString (!stdenv.cc.isClang && !stdenv.hostPlatform.isMusl) "-lgcc_s"
```

Comment:
I wonder if it wouldn't be better to do a positive check only for the platform that needs this flag (GCC + glibc?).

Reply:
```suggestion
  NIX_LDFLAGS = lib.optionalString stdenv.cc.isGNU "-lgcc_s"
```

Reply:
I updated the PR, tell me what you think.

Reply:
Ah, perfect!